### PR TITLE
Add more unit tests

### PR DESCRIPTION
**Pull request by cfreksen**
This adds more unit tests (and fixes a broken one) so that almost all code is covered.

### CODE REVIEW

**Comment by joshcanhelp** [comment]
Thank you for this @cfreksen! We'll get this reviewed as soon as we can.

File: auth0/v3/test/authentication/test_base.py [hunk 1]
```diff
@@ -100,3 +100,17 @@ def test_post_error_with_no_response_text(self, mock_post):
...
+
+        mock_get.return_value.status_code = 200
+        mock_get.return_value.text = '{"x": "y"}'
+
+        result = ab.get('the-url', params={'a': 'b'}, headers={'c': 'd'})
+
+        mock_get.assert_called_with(url='the-url', params={'a': 'b'},
+                                    headers={'c': 'd'})
+
+        self.assertEqual(result, '{"x": "y"}')
```
Comment (joshcanhelp) [inline auth0/v3/test/authentication/test_base.py]: Should this also assert the status code?

File: auth0/v3/test/management/test_rest.py [hunk 2]
```diff
@@ -198,6 +198,56 @@ def test_post_error_with_no_response_text(self, mock_post):
             self.assertEqual(context.exception.error_code, 'a0.sdk.internal.unknown')
             self.assertEqual(context.exception.message, '')
 
+    @mock.patch('requests.post')
+    def test_post_error_with_error_but_no_error_code(self, mock_post):
+        rc = RestClient(jwt='a-token', telemetry=False)
+
+        for error_status in [400, 500, None]:
+            mock_post.return_value.status_code = error_status
+            mock_post.return_value.text = '{"error": "SomeError"}'
```
Comment (joshcanhelp) [inline auth0/v3/test/management/test_rest.py]: Our errors typically come back with both an `error` property and an `error_description` one. I don't know OTOH but I'm guessing the `exception` that's returned will include the description as an error message. Might be good to check for that as well.

File: auth0/v3/test/management/test_jobs.py [hunk 3]
```diff
@@ -17,11 +17,11 @@ def test_get(self, mock_rc):
         )
 
     @mock.patch('auth0.v3.management.jobs.RestClient')
-    def get_failed_job(self, mock_rc):
+    def test_get_failed_job(self, mock_rc):
```
Comment (joshcanhelp) [inline auth0/v3/test/management/test_jobs.py]: 👍

File: auth0/v3/test/management/test_rest.py [hunk 4]
```diff
@@ -198,6 +198,56 @@ def test_post_error_with_no_response_text(self, mock_post):
...
+
+            with self.assertRaises(Auth0Error) as context:
+                rc.post('the-url')
+
+            self.assertEqual(context.exception.status_code, error_status)
+            self.assertEqual(context.exception.error_code, 'SomeError')
+
+    @mock.patch('requests.post')
+    def test_file_post(self, mock_post):
```
Comment (joshcanhelp) [inline auth0/v3/test/management/test_rest.py]: Are we doing this anywhere in the SDK?

File: auth0/v3/test/management/test_rest.py [hunk 5]
```diff
@@ -198,6 +198,56 @@ def test_post_error_with_no_response_text(self, mock_post):
...
+        files = {'file': ('some.file', 'Has some text in it\n')}
+        mock_post.return_value.status_code = 200
+        response = rc.file_post('the/url', data=data, files=files)
+        mock_post.assert_called_with('the/url', data=data,
+                                     files=files, headers=headers)
+
+        self.assertEqual(response, {'a': 'b'})
+
+    @mock.patch('requests.post')
+    def test_file_post_errors(self, mock_post):
```
Comment (joshcanhelp) [inline auth0/v3/test/management/test_rest.py]: Are we doing this anywhere in the SDK?

**Comment by lbalmaceda** [comment]
@cfreksen any extra guidance you need to complete this PR? Don't hesitate to ask any questions.

**Comment by joshcanhelp** [comment]
@cfreksen - We'd like to merge this but need a few changes. We'll leave this open for a bit longer and then we'll have to close it. Thank you!

**Pull request closed without merging**